packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -220,6 +220,12 @@ class TextEditingController extends ValueNotifier<TextEditingValue> {
   void clearComposing() {
     value = value.copyWith(composing: TextRange.empty);
   }
+
+  /// Check that the [selection] is valid, and not outside of the bounds of
+  /// [text].
+  bool isSelectionValid(TextSelection selection) {
+    return selection.start <= text.length && selection.end <= text.length;
+  }
```

Comment:
Are there any conditions that `selection.start` and `selection.end` must satisfy? Such as `selection.start >= 0` or `selection.start <= selection.end`?

I see you are extracting an existing function here; I'm just wondering if this is the correct logic.